src/pipelines/vaccinations/sk_authority.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Any, Dict, List, Callable, Optional
+
+from pandas import DataFrame, merge
+import requests
+import uuid
+import json
+
+from lib.data_source import DataSource
+from lib.time import date_range, date_today, date_offset, datetime_isoformat
+from lib.utils import table_rename
+
+
+def _download_from_api(
+    url: str, offset: Optional[int] = None, log_func: Callable[[str], None] = None
+) -> List[Dict[str, Any]]:
+    """
+    Recursively download all records from data source's API respecting the maximum record
+    transfer per request.
+    """
+    if offset:
+        url_tpl = url + "?offset={offset}"
+        url_fmt = url_tpl.format(offset=offset)
+    else:
+        url_fmt = url
+    get_opts = dict(timeout=60)
+
+    try:
+        res = requests.get(url_fmt, **get_opts)
+        if res.status_code != 200:
+            raise ConnectionError(f"Could not connect to {url}")
+        rows = res.json()
+    except Exception as exc:
+        if log_func:
+            log_func(requests.get(url_fmt, **get_opts).text)
+        raise exc
+
+    if offset is None:
+        # This is a non-recursive call.
+        return rows
+
+    if "next_offset" not in rows or rows["next_offset"] == None:
+        return rows["page"]
+    else:
+        return rows["page"] + _download_from_api(url, offset=rows["next_offset"])
```

Comment:
You are not passing `log_func` as argument (yes, this is also a bug in the other data source...)

Reply:
BTW I would pass `self.log_info` as `log_func` since `DataSource` is an instance of `ErrorLogger`.

Reply:
I resolved this to pass self.log_error (since it seemed like an error). Let me know if it should be log_info for some reason.